src/box/tuple.h

```diff
@@ -170,6 +170,26 @@ box_tuple_format(box_tuple_t *tuple);
 const char *
 box_tuple_field(box_tuple_t *tuple, uint32_t fieldno);
 
+/**
+ * Return the raw tuple field pointed by a full JSON path in
+ * MsgPack format.
+ *
+ * The JSON path includes the outmost field. For example, "c" in
+ * ["a", ["b", "c"], "d"] can be accessed using '[2][2]' path. The
+ * array element accessors are one-based (as in Lua).
+ *
+ * The first JSON path token may be a field name if the tuple
+ * has associated format with named fields.
+ *
+ * The buffer is valid until next call to box_tuple_* functions.
+ *
+ * Return NULL if the field does not exist or if the JSON path is
+ * malformed or invalid. Multikey JSON path token [*] is treated
+ * as invalid in this context.
+ */
+API_EXPORT const char *
+box_tuple_field_by_full_path(box_tuple_t *tuple, const char *path,
```

Comment:
Why "full" path? There's not and will not be a notion of a relative path in the public API. We use it only in our internal implementation for convenience and performance reasons.

Reply:
`const char *path` in `box_key_part_def_t` is relative. TBH, I don't know, whether we should reflect whether a JSON path is full/relative in a funtion name. It should be more or less obvious from the definition: if it accepts a field number, then the path is likely relative.

I'm in doubts here, but if you're sure that it worth it, I'll apply your suggestion.

Reply:
I didn't know about `box_key_part_def_t`. I wouldn't expose field+path to users in the first place - I'd rather use a union instead (the user would set either fieldno or full path specifying in flags which should be used).

Still, I think it's probably better to name this function `box_tuple_field_by_path`, because I just can't think why users would prefer to specify fieldno plus relative path instead of a full path, especially taking into account the fact that fields are usually accessed by name, not by number.

Reply:
BTW, `path` in key parts in `<space object>:index_create()` in the Lua API also starts from given field. So one need to pass `{field = <...>, path = <...>}` as a key part here. It would be nice to do just `{path = <...>}` with a full path or, if it is confusing, `{full_path = <...>}`.

I don't mind to change the function name here, but, it seems, it would be good enhance the `create_index()` signature as well.

Reply:
`create_index()` can take `path` without field, try:

```Lua
box.schema.space.create('test')
box.space.test:format({ {"foo", "map"} })
box.space.test:create_index('pk', {parts = { {"foo.bar", "unsigned"} }})
```

Reply:
Wow, it works. Sadly, it is not a canonical form: when such index object is serilized to the console, it'll show separate `fieldno` and `path` (relative).

Reply:
Renamed from `box_tuple_field_by_full_path()` to `box_tuple_field_by_path()`.